snap/hooks/post-refresh

```diff
@@ -0,0 +1,138 @@
+    exit 0
+fi
+
+# if however the previous revision wasn't set, we know that the previous 
+# revision would have used absolute paths for things like $SNAP_DATA to
+# and are now incorrect, and so we should update relevant values to use the 
+# current symlink intead
+# see https://github.com/edgexfoundry/edgex-go/issues/799#issuecomment-480318857
+# for full details
+
+# the full set of previous config items that would have been set as verbatim 
+# $SNAP_DATA with revision
+# * $SNAP_DATA/config/config-seed/res/properties/edgex-support-rulesengine/application.properties:
+#     $SNAP_DATA/support-rulesengine/rules
+#     $SNAP_DATA/support-rulesengine/templates
+# * $SNAP_DATA/config/security-secret-store/res/configuration.toml:
+#     $SNAP_DATA/vault/pki
+#     tokenfolderpath = "$SNAP_DATA/config/security-secret-store/res"
+# * $SNAP_DATA/config/security-api-gateway/res/configuration.toml:
+#     tokenpath = "$SNAP_DATA/config/security-api-gateway/res/kong-token.json"
+# * $SNAP_DATA/config/security-secret-store/vault-config.hcl
+#     tls_client_ca_file ="$SNAP_DATA/vault/pki/EdgeXFoundryCA/EdgeXFoundryTrustCA.pem"
+#     tls_cert_file ="$SNAP_DATA/vault/pki/EdgeXFoundryCA/localhost.pem"
+#     tls_key_file = "$SNAP_DATA/vault/pki/EdgeXFoundryCA/localhost.priv.key"
+# * $SNAP_DATA/config/security-api-gateway/kong.conf
+#     prefix = $SNAP_DATA/kong
+# * $SNAP_DATA/config/security-secret-store/pkisetup-kong.json
+#     working_dir: $SNAP_DATA/vault
+# * $SNAP_DATA/config/security-secret-store/pkisetup-vault.json
+#     working_dir: $SNAP_DATA/vault
+# * $SNAP_DATA/config/sys-mgmt-agent/configuration.toml:
+#     ExecutorPath = "$SNAP/bin/sys-mgmt-agent-snap-executor.sh"
+
+# snap setup dirs
+SNAP_DATA_CURRENT=${SNAP_DATA/%$SNAP_REVISION/current}
+SNAP_CURRENT=${SNAP/%$SNAP_REVISION/current}
+SNAP_DATA_DIR=$(dirname "$SNAP_DATA")
+SNAP_DIR=$(dirname "$SNAP")
+
+RULES_ENGINE_PROP_FILE="$SNAP_DATA/config/config-seed/res/properties/edgex-support-rulesengine/application.properties"
+if [ -f "$RULES_ENGINE_PROP_FILE" ]; then
+    sed -i \
+        -e "s@$SNAP_DATA_DIR/[0-9]*/support-rulesengine/rules@$SNAP_DATA_CURRENT/support-rulesengine/rules@" \
+        -e "s@$SNAP_DATA_DIR/[0-9]*/support-rulesengine/templates@$SNAP_DATA_CURRENT/support-rulesengine/templates@" \
+        "$RULES_ENGINE_PROP_FILE"
+fi
+
+SECRET_STORE_CONFIG="$SNAP_DATA/config/security-secret-store/res/configuration.toml"
+if [ -f "$SECRET_STORE_CONFIG" ]; then
+    sed -i \
+        -e "s@$SNAP_DATA_DIR/[0-9]*/vault/pki@$SNAP_DATA_CURRENT/vault/pki@" \
+        -e "s@$SNAP_DATA_DIR/[0-9]*/config/security-secret-store/res@$SNAP_DATA_CURRENT/config/security-secret-store/res@" \
+        "$SECRET_STORE_CONFIG"
+fi
+
+SYS_MGMT_AGENT_CONFIG="$SNAP_DATA/config/sys-mgmt-agent/res/configuration.toml"
+if [ -f "$SYS_MGMT_AGENT_CONFIG" ]; then
+    sed -i \
+        -e "s@$SNAP_DIR/[0-9]*/bin/sys-mgmt-agent-snap-executor.sh@$SNAP_CURRENT/bin/sys-mgmt-agent-snap-executor.sh@" \
```

Comment:
Wasn't the executor added after Delhi was released?

Reply:
Yes, you're right I misapplied the commit from edinburgh where I originally wrote this commit. I will remove it.